src/main/java/com/microsoft/graph/core/tasks/LargeFileUploadTask.java

```diff
@@ -275,9 +277,12 @@ private long nextSliceSize(long rangeBegin, long rangeEnd)  {
         long size = rangeEnd - rangeBegin + 1;
         return Math.min(size, this.maxSliceSize);
     }
-    private byte[] chunkInputStream(InputStream stream, int begin, int length) throws IOException {
-        byte[] buffer = new byte[length];
-        int lengthAssert = stream.read(buffer, begin, length);
```

Comment:
I think the IndexOutOfBoundsException may be getting thrown because [`read`](https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#read-byte:A-int-int-) starts writing to the buffer at index `begin`, which may not exist after the first slice is uploaded.

Using [`read`](https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#read-byte:A-) might be the fix?

```java
int lengthAssert = stream.read(buffer);
```